### PR TITLE
Remove fireOnce argument from set.Watch()

### PR DESCRIPTION
**Pull request by anthonybishopric**
Signed-off-by: Anthony Bishopric git@anthonybishopric.com

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -309,7 +303,7 @@ func (s *serviceSet) Watch(bringCurrent bool, fireOnce bool) chan *agent.Service
 	} else {
 		updates = make(chan *agent.ServiceUpdate)
 	}
-	s.watches[updates] = fireOnce
+	s.watches[updates] = true
```
Comment (titanous) [inline client.go]: The `watches` map value type may now be changed to a `struct{}`, which consumes zero memory.

**Comment by titanous** [comment]
LGTM. Squash, and I'll merge.

**Comment by titanous** [comment]
I'm running the tests locally and they are hanging consistently: https://gist.github.com/titanous/4686dcbf33e01b5839a1

**Comment by titanous** [comment]
It seems likely that this is a bug surfaced by removing the 1ms timeout.

**Comment by anthonybishopric** [comment]
I'll add it back in for now and make a separate commit to figure out why it's happening

**Comment by anthonybishopric** [comment]
confirmed that removing the 1ms timeout causes hanging

**Comment by titanous** [comment]
Yeah, I'm digging into TestFiltering to find the bug. Let's just do the Watch signature change here.

**Comment by titanous** [comment]
Fixed in 7235c1f2d841a601bb222848495cba2b2491d51a, drop the last commit and we should be good to go.

**Comment by titanous** [comment]
Thanks!

**Pull request merged**